enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/LeaderListener.java

```diff
@@ -22,7 +22,7 @@
 public interface LeaderListener
 {
     void onLeaderSwitch( LeaderInfo leaderInfo );
-    default void onLeaderStepDown( LeaderInfo leaderInfo )
+    default void onLeaderStepDown( long stepDownTerm )
```

Comment:
Would be good to document that leader step down must be called before leader switch, since I believe we rely on that assumption. Perhaps they should switch order in this interface then as well.

Reply:
I'll do you one better - @andrewkerr9000 suggested actually taking this ordering out of the hands of the implementing class. Incoming